test/integration/builder/main.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/go-openapi/spec"
+	"k8s.io/kube-openapi/pkg/builder"
+	"k8s.io/kube-openapi/pkg/common"
+	"k8s.io/kube-openapi/test/integration/pkg/generated"
+)
+
+const default_swagger_file = "generated.json"
+
+func main() {
+	// Get the name of the generated swagger file from the args
+	// if it exists; otherwise use the default file name.
+	swagger_filename := default_swagger_file
+	if len(os.Args) > 1 {
+		swagger_filename = os.Args[1]
+	}
+
+	// Generate the definition names from the map keys returned
+	// from GetOpenAPIDefinitions.
+	var def_names []string
+	for name, _ := range generated.GetOpenAPIDefinitions(fakeRefCreation) {
+		def_names = append(def_names, name)
+	}
+
+	// Create a minimal builder config, then call the builder with the definition names.
+	config := createOpenAPIBuilderConfig()
+	config.GetDefinitions = generated.GetOpenAPIDefinitions
+	swagger, serr := builder.BuildOpenAPIDefinitionsForResources(config, def_names...)
+	if serr != nil {
+		panic(fmt.Sprintf("ERROR: %s", serr.Error()))
+	}
+
+	// Marshal the swagger spec into JSON, then write it out.
+	specBytes, err := json.MarshalIndent(swagger, " ", " ")
+	if err != nil {
+		panic(fmt.Sprintf("json marshal error: %s", err.Error()))
+	}
+	err = ioutil.WriteFile(swagger_filename, specBytes, 0644)
+	if err != nil {
+		panic(fmt.Sprintf("stdout write error: %s", err.Error()))
+	}
+}
+
+// CreateOpenAPIBuilderConfig hard-codes some values in the API builder
+// config for testing.
+func createOpenAPIBuilderConfig() *common.Config {
+	return &common.Config{
+		ProtocolList:   []string{"https"},
+		IgnorePrefixes: []string{"/swaggerapi"},
+		Info: &spec.Info{
+			InfoProps: spec.InfoProps{
+				Title:   "Integration Test",
+				Version: "1.0",
+			},
+		},
+	}
+}
+
+// Passed into GetOpenAPIDefinitions, but unused.
+func fakeRefCreation(name string) spec.Ref {
```

Comment:
Nit: since it does nothing and isn't used anywhere else, you might define inline (where it's used).